### DISCUSSION TITLE
How to avoid setting the same background image for all the slides by ![bg](image_file_path) ?

### DISCUSSION THREAD

**Discussion by Minyus** [marp-team/marp#67] [Q&A] [answered]
Thank you for the awesome tool!

I'm trying to use a background image for all the slides.
Is there any smarter way to avoid repeating `![bg](image_image_path)` in all the slides?

**Comment by yhatt** [comment, marked as answer]
Try to use [`backgroundImage` local directive](https://marpit.marp.app/directives?id=backgrounds).

```markdown
---
backgroundImage: url('image_file_path')
---

your contents here
```

There is an exactly matched example at https://marp.app/!

**Reply by Minyus** [reply]
Worked perfect! Thank you!

**Reply by jmrcardoso** [reply]
Thanks. Any chance of setting up opacity for the global image background?

**Reply by yhatt** [reply]
@jmrcardoso 

> So, first, the bad news…
>
> **There’s no CSS property that you can use to change the opacity of only the background image.**
>
> _&mdash; https://coder-coder.com/background-image-opacity/_

Instead of setting opacity to the background image, you also can fill by a single color image with a specific transparency over the background. `background-image` property allows multiple backgrounds and CSS `linear-gradient` function is treated as the image, so the definition is:

```markdown
---
backgroundImage: linear-gradient(rgba(0 0 0 / 50%),rgba(0 0 0 / 50%)), url('./your-image.png')
---

your contents here
```

**Reply by jmrcardoso** [reply]
Thanks. Works perfectly!

**Comment by anthonio9** [comment]
Hi, is there a way to set the same background to a range of slides instead of all of them? That'd be very helpful as well :)

**Reply by yhatt** [reply]
Considering [the spec of directives](https://marpit.marp.app/directives?id=diagram), you can define a directive on the first slide to enable the background image and another directive on the slide following the end of the range to remove it.

```markdown
# 1st slide

---

<!-- backgroundImage: url('image_file_path') -->

# 2nd slide

---

# 3rd slide

---

<!-- backgroundImage: none -->

# 4th slide
```

This example means to set the same background from 2 to 3 page.
